### PR TITLE
test: add all solc latest revisions in unit tests

### PR DESCRIPTION
**Pull request by antonbaliasnikov**
# What ❔

Test all solc latest revisions in unit tests.

## Why ❔

To catch bugs with earlier solc versions.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `cargo fmt` and checked with `cargo clippy`.

### CODE REVIEW

File: .github/actions/install-solc/action.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ runs:
       run: |
         mkdir -p ${{ inputs.output-dir }}
         for SOLC_VERSION in ${{ inputs.solc-versions }}; do
-          OUTPUT="${{ inputs.output-dir }}/solc-${SOLC_VERSION}"
+          OUTPUT="${{ inputs.output-dir }}/solc-${SOLC_VERSION%-1.0.1}"
```
Comment (hedgar2017) [inline .github/actions/install-solc/action.yml:22]: @antonbaliasnikov maybe you have a better idea 😅

File: src/tests/factory_dependency.rs [hunk 2]
```diff
@@ -7,11 +7,37 @@
 use std::collections::BTreeMap;
 
 use crate::solc::pipeline::Pipeline as SolcPipeline;
+use crate::solc::Compiler as SolcCompiler;
+
+#[test]
```
Comment (antonbaliasnikov) [inline src/tests/factory_dependency.rs:12]: @hedgar2017 ,

Should we think about using [`rstest` and parametrized test cases](https://crates.io/crates/rstest)? It should remove all this boilerplate code.

image

File: .github/actions/install-solc/action.yml (hunk 1 above)
Comment (antonbaliasnikov) [reply .github/actions/install-solc/action.yml:22]: @hedgar2017 , I believe we can use `${SOLC_VERSION%%-*}` to filter out everything after `-` that will not require any update when we move to 1.0.2, etc. in the future.

Comment (hedgar2017) [reply .github/actions/install-solc/action.yml:22]: I believe we do global replacement, but a good point!

File: src/tests/factory_dependency.rs (hunk 2 above)
Comment (hedgar2017) [reply src/tests/factory_dependency.rs:12]: Yes, let's try this in another PR.
I'm wary about macro solutions in Rust, as they're basically developed from scratch and you may run into sort of JS-style DevEx :)

**Comment by hedgar2017** [comment]
Need to implement a simple AST check that would solve this cryptic error from the EVM assembly pipeline.
Will do later today.

**Pull request merged**